scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -779,27 +779,15 @@ private void receiveNextDatagramFromNetwork(DatagramPacket packet) throws IOExce
 		for (final Record record : records) {
 			try {
 
-				switch (record.getType()) {
-				case HANDSHAKE:
-				case APPLICATION_DATA:
-				case ALERT:
-				case CHANGE_CIPHER_SPEC:
-
```

Comment:
Maybe it was not so bad to remove discard record before to create a new "task" ?

Reply:
The bad's are already removed in  `List<Record> Record.fromByteArray` by

```
ContentType contentType = ContentType.getTypeByValue(type);
if (contentType == null) {
	LOGGER.debug("Received DTLS record of unsupported type [{}]. Discarding ...", type);
} else {
	records.add(new Record(contentType, version, epoch, sequenceNumber, fragmentBytes, peerAddress));
}
```